api/src/test/java/org/openmrs/validator/AllergyValidatorTest.java

```diff
@@ -178,4 +178,18 @@ public void validate_shouldPassForAValidAllergy() {
 		
 		assertFalse(errors.hasErrors());
 	}
+	
+	@Test
+	public void valid_shouldRejectNumericAllergenValue() {
+		
+		Allergy allergy = new Allergy();
+		Allergen allergen = new Allergen(AllergenType.DRUG, null , "sometxt" );
+		allergy.setAllergen(allergen);
+		allergy.getAllergen();
```

Comment:
Do you need this line?

Reply:
Yes... @HerbertYiga ..........so that the **allergen.class** is called in the **Allergy.class**

Reply:
I'm with @HerbertYiga on this. I don't really see what calling `allergy.getAllergen()` does here... all it does is return the `allergen` you just set by using `allergy.setAllergen()` and that result is ignored, so this is effectively dead code.

Otherwise, this looks fine. Thanks @Muta-Jonathan.

Reply:
Thanks, @ibacher ........have removed `allergy.getAllergen();`.........
@HerbertYiga have done the needful